docs/aws/audit/kubernetesmonitoring/rules/sg_secure_only.mdx

```diff
@@ -23,6 +23,266 @@ SOC2, GDPR, PCIDSS, NIST, HITRUST, NISTCSF
+
+2. Once the AWS CLI is installed and configured, you can use the following command to list all the security groups associated with your EKS cluster:
+
+   ```
+   aws eks describe-cluster --name <cluster-name> --query "cluster.resourcesVpcConfig.securityGroupIds"
+   ```
+   Replace `<cluster-name>` with the name of your EKS cluster. This command will return the IDs of the security groups associated with your EKS cluster.
+
+3. Now, you can use the following command to describe the inbound rules of a security group:
+
+   ```
+   aws ec2 describe-security-groups --group-ids <security-group-id>
+   ```
+   Replace `<security-group-id>` with the ID of the security group you want to check. This command will return the inbound rules of the specified security group.
+
+4. Finally, you need to check the output of the previous command. Look for the inbound rules that allow traffic on port 443. If there are any inbound rules that allow traffic on other ports, then the EKS cluster is misconfigured.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries:
+   You will need the boto3 library to interact with AWS services. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+
+2. Set up AWS credentials:
+   You need to configure your AWS credentials to allow your script to interact with your AWS services. You can do this by setting the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables, or by using the AWS CLI's configure command:
+   ```
+   aws configure
+   ```
+
+3. Write a Python script to check the EKS cluster's security group rules:
+   Here is a simple script that lists all the security groups for an EKS cluster and checks if they only allow inbound traffic on port 443:
+
+   ```python
+   import boto3
+
+   # Create a session using your AWS credentials
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+   )
+
+   # Create an EKS client
+   eks = session.client('eks')
+
+   # Get the details of the EKS cluster
+   cluster = eks.describe_cluster(name='your-cluster-name')
+
+   # Get the security group IDs
+   security_group_ids = cluster['cluster']['resourcesVpcConfig']['securityGroupIds']
+
+   # Create an EC2 resource
+   ec2 = session.resource('ec2')
+
+   # Check the inbound rules of each security group
+   for sg_id in security_group_ids:
+       security_group = ec2.SecurityGroup(sg_id)
+       for permission in security_group.ip_permissions:
+           # If there is any rule that allows traffic on a port other than 443, print a warning
+           if permission['FromPort'] != 443 or permission['ToPort'] != 443:
+               print(f'Security group {sg_id} allows traffic on port {permission["FromPort"]}-{permission["ToPort"]}')
+   ```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>